### ISSUE TITLE
[GlobalStep] "Stock Quantity" fails to update for some products using bulk edit action.

### ISSUE THREAD

**Issue by gglobalstep** [woocommerce/woocommerce#29789]
## Bug Description:
"Stock Quantity" fails to update for some products when products imported via CSV and edited using bulk action.

## Environment:
Woocommerce Version : WooCommerce 5.2.2 and 5.3 RC 1
WordPress version: (v5.7)

**PC:**
Windows 10, Mac 10.14.6
Chrome(Version 89.0.4389.90) 
Firefox(Version 87.0)
Safari: v13.1.1

## Steps To Reproduce:
1. Create any test site using JN site.
1. Install and activate all the required plugins 
1. Complete the onboarding setup wizard setup 
1. Import the Product CSV.
1. Go To Products page.
1. Select all the products and select "Edit" option from bulk action dropdown.
1. Now Change the "Manage Stock" field to "Yes".
1. Change "Stock qty" field to "Change to/Increase exisitng stock by/Decrease existing stock by".
1. Enter the quantity say "10". Click "Update".
1. Observe that stock quantity changes for some products and fails to changes for few products.

## Actual Result:
"Stock Quantity" fails to update for some products using bulk edit action.

## Expected Result:
"Stock Quantity" should update for some products using bulk edit action.

## Video:

https://user-images.githubusercontent.com/41110392/116568192-ea9d6600-a925-11eb-9fa6-2eb99b0d358c.mp4

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

<details>
```
`
### WordPress Environment ###

WordPress address (URL): https://ephemeral-tammullen-20210426.atomicsites.blog
Site address (URL): https://ephemeral-tammullen-20210426.atomicsites.blog
WC Version: 5.3.0
REST API Version: ✔ 5.3.0
WC Blocks Version: ✔ 4.9.1
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 2.2.1
Log Directory Writable: ✔
WP Version: 5.7.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.0.3
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.75.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.4.18-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.3.0
WC Database Prefix: wp_
Total Database Size: 4.94MB
Database Data Size: 3.51MB
Database Index Size: 1.43MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_ce4wp_abandoned_checkout: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailchimp_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 2.47MB + Index: 0.03MB + Engine InnoDB
wp_postmeta: Data: 0.13MB + Index: 0.06MB + Engine InnoDB
wp_posts: Data: 0.05MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 23
customize_changeset: 1
elementor_library: 1
page: 7
post: 2
product: 18
product_variation: 7
revision: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (10) ###

Akismet Anti-Spam: by Automattic – 4.1.9
Creative Mail by Constant Contact: by Constant Contact – 1.3.5
Elementor: by Elementor.com – 3.2.2
Facebook for WooCommerce: by Facebook – 2.4.0
Jetpack by WordPress.com: by Automattic – 9.6.1
Google Ads & Marketing by Kliken: by Kliken – 1.0.7
Mailchimp for WooCommerce: by Mailchimp – 2.5.1
WooCommerce Beta Tester: by WooCommerce – 2.0.2
WooCommerce Shipping & Tax: by WooCommerce – 1.25.12
WooCommerce: by Automattic – 5.3.0-RC.1

### Inactive Plugins (7) ###

AMP: by AMP Project Contributors – 2.0.11
CoBlocks: by GoDaddy – 2.11.0
Crowdsignal Forms: by Automattic – 1.4.5
Gutenberg: by Gutenberg Team – 10.4.1
Layout Grid: by Automattic – 1.6
Page Optimize: by Automattic – 0.5.1
WordPress.com Editing Toolkit: by Automattic – 3.5358

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Must Use Plugins (1) ###

WP.com Site Helper: by  –

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 3.5.1
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 21
Oldest: 2021-04-28 10:33:48 +0000
Newest: 2021-04-29 13:32:42 +0000


### Status report information ###

Generated at: 2021-04-29 14:21:03 +00:00
`
```
</details>

**Comment by tammullen** [comment]
Hi @gglobalstep ,

Thank you for taking the time to report this bug, we really appreciate your help. I can reproduce it on my end. 

Stock quantity changes for some products and fails to changes for few products.

![Screenshot 2021-05-09 at 16 49 40](https://user-images.githubusercontent.com/24649833/117578501-95601200-b0e6-11eb-8aa5-d21f58d38b4d.png)

Adding some extra context for anyone trying to fix this bug:

Something strange I noticed when trying to reproduce this was when I looked at two simple products for which one updated and the other didn't.
They appeared to be = almost identical and only had different names and prices.

Even just trying to use Bulk edit only on the product which wasn't updating didn't work
![Screenshot 2021-05-09 at 16 19 56](https://user-images.githubusercontent.com/24649833/117578692-6d24e300-b0e7-11eb-9da0-8789dfb401fe.png)
![Screenshot 2021-05-09 at 16 20 08](https://user-images.githubusercontent.com/24649833/117578696-71510080-b0e7-11eb-8c8d-f5cd3934ef24.png)

I then noticed that all of the products that hadn't updated had an `ID` greater than 18.
![Screenshot 2021-05-09 at 16 18 56](https://user-images.githubusercontent.com/24649833/117578713-89c11b00-b0e7-11eb-8758-6faacfb6978c.png)

So I tried setting both the `ID` and `product_id` to a lower number that wasn't already in use.

![Screenshot 2021-05-09 at 16 26 52](https://user-images.githubusercontent.com/24649833/117578717-92b1ec80-b0e7-11eb-83cb-c8afa37e6278.png)
I then tried updating using bulk edit again and this time it did work
![Screenshot 2021-05-09 at 16 57 39](https://user-images.githubusercontent.com/24649833/117578738-b5dc9c00-b0e7-11eb-88b9-a81bfb5d933c.png)

I then tried reproducing this theory on a new site by adding 9 fresh simple products, the 9th product had an `ID` of 19 and was the only one not to update with bulk edit. When I changed the ID to a lower number it did update.

I notice something similar in the video in the original issue that the products with lower `ID` numbers did update and the others didn't.

We’re going to solve this bug as soon as possible. Stay tuned for updates.

**Comment by roykho** [comment]
I can't seem to replicate this issue. Here's what I did.

I started with several products that are out of stock as shown in the video above.
![](https://d.pr/i/iG773C+)

Then I selected all the products and edit. Change manage stock to `yes` and increase stock qty to `20` in my case. This is the result I got.

![](https://d.pr/i/UP716q+)

I also checked all the other products and they all got increased by `20`
